src/ResourceManager/Compute/Commands.Compute/Models/AzureDiskEncryptionStatusContext.cs

```diff
@@ -3,16 +3,41 @@
 
 namespace Microsoft.Azure.Commands.Compute.Models
 {
+    enum EncryptionStatus
+    {
+        Encrypted,
+        NotEncrypted,
+        Unknown
+    }
+
     class AzureDiskEncryptionStatusContext
     {
-        public bool OsVolumeEncrypted { get; set; }
+        public EncryptionStatus OsVolumeEncrypted { get; set; }
+        public DiskEncryptionSettings OsVolumeEncryptionSettings { get; set; }
+        public EncryptionStatus DataVolumesEncrypted { get; set; }
+        [JsonIgnore]
+        public string OsVolumeEncryptionSettingsText
+        {
+            get { return JsonConvert.SerializeObject(OsVolumeEncryptionSettings, Formatting.Indented); }
+        }
+    }
```

Comment:
Could you remove OsVolumeEncryptionSettingsText ? Powershell community does not like additional parameter only for showing contents @singhkay